### PR TITLE
Enhancement #7 implementation

### PR DESCRIPTION
**Pull request by FernandoNSC5**
Partial implementation. 
@oswaldobapvicjr, please check if it do make sense. Still missing JUnits; To be implemented later on.

### CODE REVIEW

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package net.obvj.confectory;
+
+/**
+ * A default {@link ConfigurationContainer}
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:4]: This class is intended for more than that. I would suggest something like:
"A Facade for common operations in the {@code Confectory} project."

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * @author @Dautomne (Fernando Tiannamen)
+ * @since 0.1.0
+ */
+public final class Confectory
+{
+
+    /**
+     * The initial {@link ConfigurationContainer} to be loaded as default
+     */
+    protected static ConfigurationContainer DEFAULT_CONFIGURATION_CONTAINER;
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:15]: (1) Make it private
(2) Name it with camelCase as this is not a constant. Suggested name: "globalConfigurationContainer"
(3) Also improve the Javadoc with the idea that this is the current object (not exactly the initial)

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    {
+        DEFAULT_CONFIGURATION_CONTAINER = new ConfigurationContainer();
+    }
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:19]: Instantiate directly in the declaration, for more simplicity.

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Sets an already existent {@link ConfigurationContainer} as Default
+     *
+     * @param pConfigurationContainer the default {@link ConfigurationContainer} to be set
+     */
+    public Confectory(ConfigurationContainer pConfigurationContainer)
+    {
+        DEFAULT_CONFIGURATION_CONTAINER = pConfigurationContainer;
+    }
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:29]: We want the Facade to be fully static. We don't want any instances of this class being created.

(1) Make the constructor private and throw an exception (see an example of similar classes in the util package)
(2) Create a static setter method `setDefaultContainer`
(3) Please don't use the "p" prefix on method parameters

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+     * Retrives the default {@link ConfigurationContainer}
+     *
+     * @return the default {@link ConfigurationContainer}
+     */
+    public static ConfigurationContainer container()
+    {
+        return DEFAULT_CONFIGURATION_CONTAINER;
+    }
+
+}
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:49]: Always add an empty line to the end of the file

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        DEFAULT_CONFIGURATION_CONTAINER = pConfigurationContainer;
+    }
+
+    /**
+     * Reset {@link ConfigurationContainer} configuration
+     */
+    public static void resetContainer()
+    {
+        DEFAULT_CONFIGURATION_CONTAINER = new ConfigurationContainer();
+    }
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:37]: Not needed (it's already achievable by `Confectory.container().reset()`)

**Review by oswaldobapvicjr — CHANGES_REQUESTED** [review]

**Review by oswaldobapvicjr — CHANGES_REQUESTED** [review]
Good job.
Now it's only missing the JUnits

File: confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Test
+    public void ensure_global_default_configuration_exists()
+    {
+        assertNotNull(Confectory.container());
+    }
+
+    @Test
+    public void ensure_facade_dataFetchStrategy_and_defaultNullValueProvider()
```
Comment (oswaldobapvicjr) [inline confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java:36]: (1) Please use the following naming convention for JUnit test names:
`method_testCondition_expectedResult()`
or
`method_expectedResult()` (when no condition or params apply)
For example:
`settings_confectorySettingsInstance` (here it's OK not to have the condition because the method has no params.

(2) It's not a good idea to rely on the internal attributes of the ConfectorySettings class, because they can change and we don't want this Façade test to be changed by every modification in ConfectorySettings.
So, checking that `Confectory.settings()` returns exactly the same object as `ConfectorySettings.getInstance()` is the best way to test this unit in particular.

File: confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+/**
+ * Unit tests for the {@link Confectory} class.
+ * 
+ * @author FernandoNSC (Fernando Tiannamen)
+ */
+public class ConfectoryTest
+{
+
+    ConfigurationContainer configurationContainer = new ConfigurationContainer();
```
Comment (oswaldobapvicjr) [inline confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java:21]: Because we have plans to modify the `equals()`) method of `ConfigurationContainer`, it's better to use a mock instead of an actual instance.

File: confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Test
+    public void ensure_facade_dataFetchStrategy_and_defaultNullValueProvider()
+    {
+        assertEquals(Confectory.settings().getDefaultDataFetchStrategy(), ConfectorySettings.getInstance().getDefaultDataFetchStrategy());
+        assertEquals(Confectory.settings().getDefaultNullValueProvider(), ConfectorySettings.getInstance().getDefaultNullValueProvider());
+    }
+
+    @Test
+    public void ensure_global_default_configuration_is_updated()
```
Comment (oswaldobapvicjr) [inline confectory-core/src/test/java/net/obvj/confectory/ConfectoryTest.java:43]: (1) Apply the naming convention (suggested name: `setDefatultContainer_notNull_updatedSuccessfully()`)

(2) Please also add another test case passing a null parameter
(`setDefaultContainer_null_exception()`)

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    */
+    public static ConfectorySettings settings()
+    {
+        return ConfectorySettings.getInstance();
+    }
+
+    /**
+     * Sets an already existent {@link ConfigurationContainer} as Default
+     *
+     * @param configurationContainer the default {@link ConfigurationContainer} to be set
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:47]: ...; not null

File: confectory-core/src/main/java/net/obvj/confectory/Confectory.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    public static void setDefaultContainer(ConfigurationContainer configurationContainer)
+    {
+        globalConfigurationContainer = configurationContainer;
```
Comment (oswaldobapvicjr) [inline confectory-core/src/main/java/net/obvj/confectory/Confectory.java:51]: Don't let the user pass null.
Check for nullity before (use `Objects.requireNonNull` with a descriptive message.

Also state this in the javadoc (with the @throws clause)

**Review by oswaldobapvicjr — CHANGES_REQUESTED** [review]

**Review by oswaldobapvicjr — APPROVED** [review]
